component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -830,6 +830,23 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
             }
             context.setProperty(SMSOTPConstants.CODE_MISMATCH, true);
         }
+
+        if (succeededAttempt && isLocalUser) {
+            String username = String.valueOf(context.getProperty(SMSOTPConstants.USER_NAME));
+            try {
+                String mobileNumber = SMSOTPUtils.getMobileNumberForUsername(username);
+                if (StringUtils.isEmpty(mobileNumber)) {
+                    String tenantDomain = context.getTenantDomain();
+                    Object verifiedMobileObject = context.getProperty(SMSOTPConstants.REQUEST_USER_MOBILE);
+                    if (verifiedMobileObject != null) {
+                        updateMobileNumberForUsername(context, request, username, tenantDomain);
+                    }
+                }
+            } catch (SMSOTPException e) {
+                throw new AuthenticationFailedException("Failed to get the parameters from authentication xml file. ", e);
```

Comment:
the error message does not seem relavant to the logic implemented here.. shall we make it more meaningful?

Reply:
Updated the error message